services/bonita-time-tracker/src/main/java/org/bonitasoft/engine/tracking/TimeTracker.java

```diff
@@ -28,151 +31,332 @@
+        flushEventListener.activate();
+        return true;
+    }
+    public boolean deactivateFlushEventListener(final String flushEventListenerName) {
+        final FlushEventListener flushEventListener = this.flushEventListeners.get(flushEventListenerName);
+        if (flushEventListener == null) {
+            return false;
+        }
+        flushEventListener.deactivate();
+        return true;
+    }
+    public void activateRecord(final TimeTrackerRecords activatedRecord) {
+        this.activatedRecords.add(activatedRecord);
+    }
+
+    public void deactivatedRecord(final TimeTrackerRecords activatedRecord) {
+        this.activatedRecords.remove(activatedRecord);
+    }
+
+    public Set<TimeTrackerRecords> getActivatedRecords() {
+        return Collections.unmodifiableSet(this.activatedRecords);
+    }
+
+    public void startTracking() {
+        if (!serviceStarted) {
+            if (logger.isLoggable(getClass(), TechnicalLogSeverity.WARNING)) {
+                this.logger.log(getClass(), TechnicalLogSeverity.WARNING,
+                        "Cannot start Time tracker tracking because service is not started.");
+            }
+            return;
+        }
+        startTracking = true;
+        internalStartTracking();
+    }
+
+    public void stopTracking() {
+        startTracking = false;
+        internalStopTracking();
+    }
+
+    FlushThread createFlushThread() {
+        return new FlushThread(this);
+    }
+
+
+    private void internalStartTracking() {
+        if (startTracking) {
+            if (logger.isLoggable(getClass(), TechnicalLogSeverity.WARNING)) {
+                this.logger.log(getClass(), TechnicalLogSeverity.WARNING,
+                        "Starting Time tracker tracking...");
+            }
+            if (!flushThread.isStarted()) {
+                flushThread.start();
+            }
+            for (final FlushEventListener listener : getActiveFlushEventListeners()) {
+                listener.notifyStartTracking();
+            }
+            if (logger.isLoggable(getClass(), TechnicalLogSeverity.WARNING)) {
+                this.logger.log(getClass(), TechnicalLogSeverity.WARNING,
+                        "Time tracker tracking is activated. This may not be used in production as performances may be strongly impacted.");
+            }
+        }
+    }
+
+    private void internalStopTracking() {
+        if (isTracking()) {
+            if (logger.isLoggable(getClass(), TechnicalLogSeverity.WARNING)) {
+                this.logger.log(getClass(), TechnicalLogSeverity.WARNING,
+                        "Stopping Time tracker tracking...");
+            }
+            if (flushThread.isStarted()) {
+                flushThread.interrupt();
+            }
+            for (final FlushEventListener listener : getActiveFlushEventListeners()) {
+                listener.notifyStopTracking();
+            }
+            if (logger.isLoggable(getClass(), TechnicalLogSeverity.WARNING)) {
```

Comment:
all these warnings could be factories in the methods because it's only constants that are given
this would make the code easier to read